### ISSUE TITLE
Setting a boolean value sets a string instead

### ISSUE THREAD

**Issue by tescalada** [pdf/knife-attribute#5]
Trying to set a boolean value ends up setting a string instead.

```
knife node attribute set testnode testvalue true
Successfully set node testnode attribute testvalue to: "true"
```

on line https://github.com/pdf/knife-attribute/blob/3703763a9a9e13651ccecc2bbac460b5932b3dc3/lib/knife-attribute/set.rb#L42 Chef::JSONCompat.from_json(value) only allows values where the top level is a hash or array.

Chef::Exceptions::JSON::ParseError: Top level JSON object must be a Hash or Array. (actual: TrueClass)

**Comment by pdf** [comment]
Right, that line is wrapped in a rescue though, so that if `from_json` fails, the value is used literally.  How are you seeing that exception?

**Comment by tescalada** [comment]
Sorry, I see that was confusing wording... I'm not seeing that exception when running the command. I got that exception while digging into why true was not turning into a boolean value.

**Comment by pdf** [comment]
Gotcha, yeah, I'm not entirely sure what the right thing to do here is - I suppose we could explicitly convert the strings `true` and `false` to bools, and numeric-looking strings to numbers, but it's hard to know whether we'd always be doing the right thing, and that has the potential to break existing users.

**Comment by tescalada** [comment]
My thinking was if using `-F json` with set, that it should parse the bools and numbers. You could always specify `-F json "'true'"` if you really wanted to use the string 'true'. If you don't set -F json it should keep the current parsing.

It's not documented what would happen if using -F json with set, so I would hope it wouldn't break existing users, but thats probably too optimistic of me.

If you wanted to introduce totally new options and not touch the current parsing theres the option of `knife node attribute set foo.example.org some.bool.value --true/--false`. Its unfortunate that the existing params happen to be -t and -F, that could possibly be confusing. Maybe numbers could have a -n number?

**Comment by pdf** [comment]
Yeah, those are options I've been considering, I'll have to think about it some more.

**Comment by neufeldtech** [comment]
For those that came here from a Google search looking on how to set a boolean value with this gem, this workaround should do the trick:

To set the attribute 'powered_off' to the boolean `false` at the top level of the node object, we can do this:

```
knife node attribute set mynode1.example.com . '{"powered_off": true}'
```
